exercise3.js

```diff
@@ -82,34 +115,85 @@
-
+// 7.a - Write a function to find the maximum numerical value of the given array.  
+//Get rid of any non numerical values.  Convert the strings that are numbers to an actual number data type.  
+//("one" => 1) ("1" => 1).  Use array methods to perform this task.  
+const numbers = [2,23,1,2,1,1,1,2,2.5,20,200,2000,{k:1},20000,19999,1878,140,23,4,"sk",true,true,"true-dat","nice","one","two","three","3","tea",[]];
+let max = [0];
 function maxNumber(numbers) {
-    //your code...
+    
+    for(let x = 0; x<numbers.length; x++){
+        
+        if(numbers[x]=="two"){
+            numbers[x]=2;
+        }
+                
+        else if(numbers[x]=='one'){
+            numbers[x]=1;
+        }
+        else if(numbers[x]=="three"){
+            numbers[x]=3;
+        }
+        else if(numbers[x]=="3"){
+            numbers[x]=3;
+        }
+        else if(typeof numbers[x] != 'number'){
+            numbers.splice(x,1);
+            x--;
+        }
+        else if(numbers[x]>max){
+            max = numbers[x];
+        }
+
+    }
+        
+        
+    
+    return console.log(max);
 }
 
+maxNumber(numbers);
+
 // 7.b -Write a function that sorts the given numbers array.  Allow the function to sort the array in descending order
 
 function sortNums(numbers,desc=false) {
-    //your code...
+    numbers.sort((a,b)=>a-b);
+    return console.log(numbers);
 };
+sortNums(numbers);
 
 
 
 
 /************************************************************* */
 // Problem 8:
-//add an example of at least 5 JavaScript data types to the given mapObj.  The key is the example data type, and the value is the name of the data type.  An object data type has already been set as the 1st key / val pair.
+//add an example of at least 5 JavaScript data types to the given mapObj.  
+//The key is the example data type, and the value is the name of the data type. 
+// An object data type has already been set as the 1st key / val pair.
+
+// let mapObj = new Map()
+// mapObj.set({company : "TEKsystems"},"object") ;
+
 
-const mapObj = new Map();
-mapObj.set({company : "TEKsystems"},"object");
 
 
-console.log(mapObj.has({company : "TEKsystems"}));  
-//The above console.log() statmeent returns false.  Write another console.log() statement explaining why this line of code prints false.  Refactor the code on line 106, so you can successfully check to see if {company : "TEKsystems"} exists in the mapObj.
 
-//your code...
+// console.log(mapObj.has({company : "TEKsystems"})); 
+// console.log("It prints false because it is being use incorrectly"); 
+//The above console.log() statmeent returns false.  
+//Write another console.log() statement explaining why this line of code prints false. 
+// Refactor the code on line 106, so you can successfully check to see if {company : "TEKsystems"} exists in the mapObj.
+
+const mapObj = new Map();
+mapObj.set('company', 'TEKsystems');
```

Comment:
you were supposed to set an object as the key, and the value to be the data type.